banking/klarna_kosma_integration/doctype/bank_reconciliation_tool_beta/bank_reconciliation_tool_beta.js

```diff
@@ -126,24 +136,29 @@ frappe.ui.form.on('Bank Reconciliation Tool Beta', {
 	},
 
 	bank_account: function (frm) {
-		frappe.db.get_value(
-			"Bank Account",
-			frm.doc.bank_account,
-			"account",
-			(r) => {
-				frappe.db.get_value(
-					"Account",
-					r.account,
-					"account_currency",
-					(r) => {
-						frm.doc.account_currency = r.account_currency;
-						frm.trigger("bank_statement_from_date");
-						frm.trigger("bank_statement_to_date");
-					}
-				);
-			}
-		);
+		if (frm.doc.bank_account) {
+			frappe.db.get_value(
+				"Bank Account",
+				frm.doc.bank_account,
+				"account",
+				(r) => {
+					frappe.db.get_value(
+						"Account",
+						r.account,
+						"account_currency",
+						(r) => {
+							frm.doc.account_currency = r.account_currency;
+							frm.trigger("bank_statement_from_date");
+							frm.trigger("bank_statement_to_date");
+						}
+					);
+				}
+			);
 
+			frm.events.get_bank_transactions(frm);
```

Comment:
Although this reduces work it also makes the UX a bit confusing IMO.
- Results area is auto rendered on setting the bank account. So the user doesn't need to click "Get Bank Transactions"
- Now if the user tinkers with the date, the expectation subconsciously became that I don't need to click the button as previously seen and the results area will just re-render, while that is not the case. A bit confusing.
- Additionally, this removes a lot of power from the primary button

I think let the user click on "Get Bank Transactions" to well..get bank transactions so they get some subconscious feedback that some filter has changed and thus the results change too. It'll keep things consistent.
Just my opinion again 😄

Reply:
I agree, and I would like to have this triggered on all relevant filter changes. I just didn't figure out how to avoid duplicate calls yet, when one filter triggers the other and both trigger `get_bank_transactions`. We can then replace the primary button with a secondary "Refresh" button.